### ISSUE TITLE
Question, or just a small bug?

### ISSUE THREAD

**Issue by Lonsdale201** [rtpHarry/jet-engine-repeater-values#1]
First, thank you for your amazing extension. :) 

At the moment your plugin supported, to attach media for a widget as background image? Any of widget have a backround option, and i tried to attach a media (which inside the repeater), unfortunately, only the first one is taken as a value, not the others.

Screenshots:
https://prnt.sc/23pcnd3
https://prnt.sc/23pcsm6

Thanks in advance :)

**Comment by rtpHarry** [comment]
Interesting. I have it working with multiple images in my project, but I haven't actually tested it with multiple backgrounds.

I will investigate this.

**Comment by Lonsdale201** [comment]
> Interesting. I have it working with multiple images in my project, but I haven't actually tested it with multiple backgrounds.
> 
> I will investigate this.

Yes, if i set the repeater media metabox for a simple image widget works like a charm, but if set with any widget bacgkround, not :/

**Comment by rtpHarry** [comment]
OK I've set up a test on my computer and confirmed it is doing the same bug. 

I'll need to get a php debugger up and running to see what is going on with the data. 

I will report back soon.

**Comment by rtpHarry** [comment]
Just wanted to give an update. I've spent an hour this evening and ruled out a few things but not found the cause yet.

**Comment by Lonsdale201** [comment]
> Just wanted to give an update. I've spent an hour this evening and ruled out a few things but not found the cause yet.

It's OK. It's not a huge bug after all.

**Comment by tanyabendis** [comment]
@rtpHarry Thanks so much for this plug-in! Is there any chance you will include check-box repeater fields in the near future?

**Comment by rtpHarry** [comment]
@tanyabendis this sounds like a separate issue. Could you open a new issue at the top and explain more what you're trying to achieve?

**Comment by DomZrrR** [comment]
@rtpHarry 
Hello Harry,
nice plugin you wrote there! It is helping us a lot.

Can you tell us the status of this issue? We have the same problems.
We created a listing repeater with injection, but unfortunately when we use the background of a section or column with the help of your dynamic field it shows only the first image in the repeater order. The others are not working.

We are using this in a customer project and it would be very nice, if it would work soon. Meanwhile we are thinking about alternativs but your plugin is solution number one.

If you need more informations or data just ask :) 

Thank you!

**Comment by rtpHarry** [comment]
The honest status is that I haven't found a solution to this yet.

The way I built this plugin was to extract the code that Crocoblock used to extract their repeater values in other places, and then wrap that up in a dynamic tag plugin.

So the actual heavy lifting is done by JetEngine:

https://github.com/rtpHarry/jet-engine-repeater-values/blob/777e1565a2910759a5dd1e44ff3897f052d5ee4c/dynamic-tags/jet-engine-repeater-dynamic-tag-text.php#L107

So it seems that it's a bug within JetEngine. 

I stepped through this a few times with a debugger and couldn't pinpoint where it was going wrong, and then haven't been back to it since.

One solution is to use an image widget instead of a background image - the dynamic tag code works then. Is that possible with your design?

Otherwise, I will try to look into this again but I already know that I have a backlog of client tasks that is going to take at least a few days for me to clear and has to be my priority.

**Comment by Nedo90** [comment]
Hi Harry, 

First of all I would like to thank you for the great plugin. I have been waiting for this for a long time. 

I have the same problem now that I get only one background image displayed. In other widgets and the containers it does not work.  I can't recreate the design with the image widget unfortunately. Have you already taken a closer look at the bug? Is there a solution?

Thank you!
![repeater](https://user-images.githubusercontent.com/86856721/201684528-52fae106-582f-4d92-a93c-a6f0c6f9eba7.png)
